tests/test_templatetags_templating.py

```diff
@@ -792,6 +792,114 @@ def test_component_nesting_deep_slot_inside_component_fill(self):
         """
         self.assertHTMLEqual(rendered, expected)
 
+    # This test is based on real-life example.
+    # It ensures that deeply nested slots in fills with same names are resolved correctly.
+    # It also ensures that the component_vars.is_filled context is correctly populated.
+    @parametrize_context_behavior(["django", "isolated"])
+    def test_component_nesting_deep_slot_inside_component_fill_2(self):
```

Comment:
Test based on my work project layout.

Annoyingly, there was one more issue with making the slots/fills and blocks/extends all work correctly, that I experienced on my website, but I couldn't figure out how to isolate it into a test.